tests/utils/fakeredis_ext.py

```diff
@@ -68,11 +70,23 @@ def _parse_command_response_from_connection(self, connection, type):
         return self._parse_command_response(cmd, args)
 
     def _parse_command_response(self, cmd, args):
-        cmd_func = getattr(self, cmd.lower())
-        return cmd_func(*args)
+        """TODO (JP) I'm not 100% sure why we are overriding the parse_response code on this class (which is what
+        ultimately leads us to here) but after moving to a newer version of fakeredis, our old code here would
+        cause an "RuntimeError: maximum recursion depth exceeded" error because cmd_func would lead us right back
+        to this method again.
 
-    def pipeline(self, transaction=True):
-        return FakePipelineWithStack(self, transaction)
+        Using a new FakeRedis client (which will _not_ call _parse_command_response) seems to work but there is
+        probably a better solution to this problem.
+
+        I'm also unsure why ZADD needs to be modified but it probably has to do with some change in the FakeRedis code
+        that we are overriding here.
+        """
```

Comment:
When I originally wrote this, `fakeredis` didn't support `zadd` so this whole mechanism was the hacky way to extend the fakeredis protocol to support `zadd`.

Reply:
Oooo wait... ignore that. I am dumb...

Ok so to explain this whole monkey patching:

Kombu uses `redis-py` in a way that it wasn't originally intended to implement an asynchronous (non-blocking) API, presumably for performance reasons. The asynchronous API makes heavy use of the lower level functionalities related to connection and request/response management in `redis-py`. Naturally, these functions were not implemented in `fakeredis` because noone really uses `redis-py` in that way.

This whole class extended the `FakeStrictRedis` class to implement stub versions of those low level functions, which were cobbled together based on the actual `redis-py` source and how kombu uses it.

Obviously, that doesn't answer why ZADD needs to be modified so my original answer is useless.

Reply:
1. Oh, hey. It's you! 
2. I forgot this repo is public.
3. Thanks for the background info on this!
4. We should get lunch sometime.